java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/websocket/GenericMqttWebSocketClient.java

```diff
@@ -0,0 +1,73 @@
+package com.cumulocity.generic.mqtt.client.websocket;
+
+import com.cumulocity.generic.mqtt.client.GenericMqttMessageListener;
+import com.cumulocity.generic.mqtt.client.converter.GenericMqttMessageConverter;
+import com.cumulocity.generic.mqtt.client.model.GenericMqttMessage;
+import org.java_websocket.client.WebSocketClient;
+import org.java_websocket.handshake.ServerHandshake;
+
+import java.net.URI;
+import java.util.Optional;
+
+class GenericMqttWebSocketClient extends WebSocketClient {
+
+    private final GenericMqttMessageConverter genericMqttMessageConverter = new GenericMqttMessageConverter();
+
+    private GenericMqttMessageListener genericMqttMessageListener;
+
+    public GenericMqttWebSocketClient(URI serverUri) {
```

Comment:
nitpick: unify constructors. That is change 'super(serverUri)' to 'this(serverUri, null)'

Maybe this is opinion

Reply:
suggestion: Looking at all the null checks in the callback methods, I'd use a (private, static) null/empty implementation of the GenericMqttMessageListener interface her instead of null or not setting one.

Reply:
This class is not exposed to the users and is internal

Reply:
Not sure if that is relevant.
1. A null listener implementation removes numerous if-statements from the code. 
2. We should be able to make the listener final (we can assign null or null implementation in the 1-arg constructor).
3. Why not reuse our own constructors: the the 1-arg constructor can pass null or null-implementation to the other constructor, which will be the only place the assignments happen.

The above suggestion each make it much neater IMO

Reply:
I would suggest to use EMPTY_LISTENER which is simply doing nothing and use it instead of null, then we can also remove those null checks.

Reply:
Done.